core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -168,82 +183,72 @@ class RecordInsightsLOCO[T <: Model[T]]
     }
   }
 
+  private def aggregateDiffs(
+    featureSparse: SparseVector,
+    aggIndices: Array[(Int, Int)],
+    strategy: VectorAggregationStrategy,
+    baseScore: Array[Double],
+    featureSize: Int
+  ): Array[Double] = {
+    strategy match {
+      case VectorAggregationStrategy.Avg =>
+        aggIndices
+          .map { case (i, oldInd) => computeDiff(featureSparse.copy.updated(i, oldInd, 0.0), baseScore) }
+          .foldLeft(Array.empty[Double])(sumArrays)
+          .map( _ / featureSize)
+
+      case VectorAggregationStrategy.LeaveOutVector =>
+        val copyFeatureSparse = featureSparse.copy
+        aggIndices.foreach {case (i, oldInd) => copyFeatureSparse.updated(i, oldInd, 0.0)}
```

Comment:
why are you using `updated` here?

Reply:
I think this should be `aggIndices.foreach {case (i, oldInd) => copyFeatureSparse.update(i, oldInd, 0.0)}`

Reply:
The idea is to zero out the entire vector and then calculate the loco in case of `VectorAggregationStrategy.LeaveOutVector`

Reply:
And in case of `VectorAggregationStrategy.Avg` the idea is to calculate the loco by zeroing out each column of the date/text feature vector and then average out all the locos calculated.

Reply:
well, `updated` returns the updated array. I think you want to modify in place (see my previous comment)

Reply:
I think it does modify in place and returns the same array. Below the definition of that method -

```implicit class RichSparseVector(val v: SparseVector) extends AnyVal {
    def updated(index: Int, indexVal: Int, value: Double): SparseVector = {
      require(v.indices(index) == indexVal,
        s"Invalid index: indices($index)==${v.indices(index)}, expected: $indexVal")
      v.values(index) = value
      v
    }
  }```